frontend/src/components/phones/dashboard/Dashboard.tsx

```diff
@@ -172,11 +188,133 @@ export const PhoneDashboard = () => {
+        moment(b.last_inbound_date).unix() - moment(a.last_inbound_date).unix()
+    )
+    .map((data) => {
+      return (
+        <li
+          key={data.id}
+          className={data.blocked ? styles["greyed-contact"] : ""}
+        >
+          <span className={styles["sender-number"]}>
+            {formatPhoneNumberToUSDisplay(data.inbound_number)}
+          </span>
+          <span
+            className={`${styles["sender-date"]} ${styles["sender-date-wrapper"]}`}
+          >
+            {data.last_inbound_type === "text" && (
+              <ForwardedTextIcon
+                alt="Last received a text"
+                className={styles["forwarded-type-icon"]}
+                width={20}
+                height={12}
+              />
+            )}
+            {data.last_inbound_type === "call" && (
+              <ForwardedCallIcon
+                alt="Last received a call"
+                className={styles["forwarded-type-icon"]}
+                width={20}
+                height={15}
+              />
+            )}
+            <Moment calendar={calendarStrings}>{data.last_inbound_date}</Moment>
+          </span>
+          <span className={styles["sender-controls"]}>
+            <button
+              onClick={() =>
+                inboundContactData.setForwardingState(!data.blocked, data.id)
+              }
+              className={styles["block-btn"]}
+            >
+              {data.blocked ? "Unblock" : "Block"}
+            </button>
+          </span>
+        </li>
+      );
+    });
+
+  const callerSMSSendersPanel = (
+    <div id="secondary-panel" className={styles["dashboard-card"]}>
+      <div className={styles["dashboard-card-caller-sms-senders-header"]}>
+        <span>
+          <button
+            type="button"
+            onClick={toggleSendersPanel}
+            className={styles["caller-sms-logs-back-btn"]}
+          >
+            <ChevronLeftIcon
+              alt="Back to Primary Dashboard"
+              className={styles["nav-icon"]}
+              width={20}
+              height={20}
+            />
+          </button>
+        </span>
+        <span className={styles["caller-sms-logs-title"]}>
+          {l10n.getString("phone-dashboard-senders-header")}
+        </span>
+        <span></span>
       </div>
+      <ul className={styles["caller-sms-senders-table"]}>
+        <li className={styles["greyed-contact"]}>
+          <span>
+            {l10n.getString("phone-dashboard-sender-table-title-sender")}
+          </span>
+          <span>
+            {l10n.getString("phone-dashboard-sender-table-title-activity")}
+          </span>
+          <span>
+            {l10n.getString("phone-dashboard-sender-table-title-action")}
+          </span>
+        </li>
+        {inboundContactArray}
+      </ul>
+    </div>
+  );
+
+  return (
+    <main className={styles["main-phone-wrapper"]}>
+      {showingPrimaryDashboard ? (
+        // Primary Panel
+        <>{primaryPanel}</>
```

Comment:
question (non-blocking): Why are we using fragments to hold `primaryPanel` - same question with ln 317 `callerSMSSendersPanel`